### PR TITLE
Fix wrong permissions when installation with enrollment on Windows in privileged mode

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Fixes an installation issue on Windows in privileged mode where the wrong permissions are given to the Elastic Agent installation directory.

## Why is it important?

Otherwise it would allow any non-Administrator to read/modify/delete a file in the Elastic Agent installation directory.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Disruptive User Impact

Fixes a disruptive user issue of invalid permissions.

## How to test this PR locally

1. Install without `--unprivileged` and with enrollment.
2. Verify the permissions of the installation directory only have access by Administrators.

## Related issues

- Closes https://github.com/elastic/ingest-dev/issues/3339

### CODE REVIEW

File: internal/pkg/agent/perms/windows.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func FixPermissions(topPath string, opts ...OptFunc) error {
 
 	// user gets grant based on the mask
 	var userSID *windows.SID
-	if o.ownership.UID != "" {
+	if o.mask&0700 != 0 && o.ownership.UID != "" {
```
Comment (cmacknz) [inline internal/pkg/agent/perms/windows.go:45]: What does the 0700 mask mean for Windows? Same thing as for Unix?

**Review by cmacknz — COMMENTED** [review]
Can we add a test to prevent a regression like this again?

Something like using an unprivileged shell/terminal to try to access/modify/delete a file in the installation directory?

**Comment by cmacknz** [comment]
This needs to be merged before the 16th to make the final build candidate, so we have some time to add the necessary validation. If it turns out we need longer than that to add validation, we can merge this first.

I've also tracked this issue as blocker on the release issue so it can't be forgotten.

**Comment by blakerouse** [comment]
Sounds good. Working on test.

**Review by cmacknz — APPROVED** [review]
Approving to avoid blocking the final 8.14 BC. Tests will follow separately.

**Pull request merged**